core/Piwik.php

```diff
@@ -321,7 +321,7 @@ public static function isUserIsAnonymous()
     {
         $currentUserLogin = Piwik::getCurrentUserLogin();
         $isSuperUser = self::hasUserSuperUserAccess();
-        return !$isSuperUser && $currentUserLogin == 'anonymous';
+        return !$isSuperUser && $currentUserLogin && strtolower($currentUserLogin) == 'anonymous';
```

Comment:
not really needed but might avoid some random case where anonymous user was not detected correctly in case someone passes eg `anOnymous` to an API

Reply:
👍

Reply:
Found one more place that might need to use Piwik::isUserIsAnonymous():
* LastSeenTimeLogger::logCurrentUserLastSeenTime()

There some other places that use `'anonymous'` directly but don't do a strtolower, not sure if any of them need to change to ensure consistency.

Reply:
will update 👍  I think looked over all/most of them and updated where needed.